default/AI/ShipDesignAI.py

```diff
@@ -0,0 +1,1602 @@
+"""
+This module deals with the autonomous shipdesign from the AI. The design process is class-based.
+The basic functionality is defined in the class AIShipDesign. The more specialised classes mostly 
+implement the rating function and some additional information for the optimizing algorithms to improve performance. 
+
+Example usage of this module:
+import ShipDesignAI
+myDesign = ShipDesignAI.AIShipDesign_Military()
+myDesign.additional_specifications.enemy_mine_dmg = 10
+best_military_designs = myDesign.optimize_design()  # best designs per planet: (rating,planetID,design_id,cost) tuples
+
+
+Available ship classes:
+- AIShipDesign_Military: basic military ship
```

Comment:
My biggest concern was the base class which I think is not wise to call "ShipDesign". Maybe something like BaseDesign would work though.

Another thing to consider is that the naming convention should be convenient to use with multiple inheritance. While my current naming style may be not too elegant and maybe not too pythonic (underscores in classnames) either, I personally think it is convenient for finding the relevant classname though that is obviously heavily biased by me thinking of that names...

Still, I will give it some more thought. If you have a coherent naming system in mind, it would be nice to supply (most of) the resulting names so we see if it works consistently in all cases.